src/presence_analyzer/tests.py

```diff
@@ -354,6 +355,30 @@ def test_group_start_end_by_weekday(self):
 
         self.assertEqual(result, utils.group_start_end_by_weekday(fake_data))
 
+    def test_cache_decorator(self):
+        """
+        Test utils/cache result
+        """
+        moc = MagicMock()
+        _func = lambda _: moc()
+        cachelimit = 0.001
+
+        wrapped = utils.cache(cachelimit)(_func)
+
+        wrapped(1)
+        wrapped(2)
+        self.assertEqual(moc.call_count, 2)
+
+        wrapped(1)
+        self.assertEqual(moc.call_count, 2)
+
+        now = datetime.datetime.now()
+        with patch('datetime.datetime') as dt:
+            dt.now.return_value = now + datetime.timedelta(seconds=2*cachelimit)
```

Comment:
we can set `cachelimit` to 10 and `seconds=` to 60.
This will make it safe for machines under load / with slower CPUs (things like tests tends to be run automatically by some CI system, those are usually under high load and tends to lag).